configuration/NI.Analysis.targets

```diff
@@ -0,0 +1,95 @@
+
+		<CodeAnalysisRuleSet>$(CodeAnalysisRuleSetDirectory)\NI.ruleset</CodeAnalysisRuleSet>
+		<CodeAnalysisRuleSet Condition="'$(IsNITestProjectCondition)' == 'True'">$(CodeAnalysisRuleSetDirectory)\NI.Tests.ruleset</CodeAnalysisRuleSet>
+		<CodeAnalysisRuleSet Condition="'$(IsNITestUtilitiesProjectCondition)' == 'True'">$(CodeAnalysisRuleSetDirectory)\NI.TestUtilities.ruleset</CodeAnalysisRuleSet>
+	</PropertyGroup>
+
+  <PropertyGroup>
+    <!-- Enable Roslyn analyzers if they haven't been explicitly disabled, we're using Roslyn for CA/SA, the code needs to be validated -->
+    <EnableRoslynAnalyzers>True</EnableRoslynAnalyzers>
+    <IncludeStyleAnalyzers>True</IncludeStyleAnalyzers>
+    <IncludeCodeAnalyzers>True</IncludeCodeAnalyzers>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <StyleCopConfig>$(CodeAnalysisRuleSetDirectory)\stylecop.json</StyleCopConfig>
+    <NI1006_BannedMethods>$(CodeAnalysisRuleSetDirectory)\BannedMethods.xml</NI1006_BannedMethods>
+    <NI1004_LiteralExemptions>$(CodeAnalysisRuleSetDirectory)\LiteralExemptions.xml</NI1004_LiteralExemptions>
+
+    <!-- TODO: Determine whether we want to use the solution directory, the project directory, or both?-->
+    <NI1704_GlobalExceptions>$(SolutionDir)\CodeAnalysisDictionary.xml</NI1704_GlobalExceptions>
+
+    <!--AdditionalSpellingDictionary.dic contains the differences in words between the hunspell and Microsoft dictionaries-->
+    <NI1704_AdditionalSpellingDictionary>$(CodeAnalysisRuleSetDirectory)\NI1704_AdditionalSpellingDictionary.dic</NI1704_AdditionalSpellingDictionary>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(IncludeStyleAnalyzers)' == 'True'">
+    <!-- By adding the Link attribute, these files will not be shown in the Solution Explorer for .NET Core projects. -->
+    <AdditionalFiles Include="$(StyleCopConfig)" Condition="Exists('$(StyleCopConfig)')">
+      <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+
+    <!-- Microsoft's SA analyzers -->
+    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
+    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll"/>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IncludeCodeAnalyzers)' == 'True'">
+    <RoslynAnalyzer Include="Microsoft.CodeQuality" />
+    <RoslynAnalyzer Include="Microsoft.NetCore" />
+    <RoslynAnalyzer Include="Microsoft.NetFramework" />
+    <RoslynAnalyzer Include="Text" />
+
+    <!-- Microsoft's CA analyzers -->
+    <Analyzer Include="@(RoslynAnalyzer->'$(NuGetPackagesDirectory)\%(Identity).analyzers\2.6.3\analyzers\dotnet\cs\%(Identity).Analyzers.dll')"/>
+    <Analyzer Include="@(RoslynAnalyzer->'$(NuGetPackagesDirectory)\%(Identity).analyzers\2.6.3\analyzers\dotnet\cs\%(Identity).CSharp.Analyzers.dll')"/>
+    <Analyzer Include="$(PkgMicrosoft_VisualStudio_Threading_Analyzers)\analyzers\cs\Microsoft.VisualStudio.Threading.Analyzers.dll"/>
```

Comment:
Where is `PkgMicrosoft_VisualStudio_Threading_Analyzers`/`PkgMicrosoft_CodeQuality_Analyzers` defined?

Reply:
I'm removing this file for now, but those variables are generated with <GeneratePathProperty> in the package references that include those packages. We'll revisit this one when we attempt to add it.